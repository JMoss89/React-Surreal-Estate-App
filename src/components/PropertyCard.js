import React from 'react';
import '../styles/PropertyCard.css';
import PropTypes from 'prop-types';

function PropertyCard(props) {
    const { title, type, bathrooms, bedrooms, price, city, email } = props;

    return (
        <div className='property-card'>
            {title}
            <br />
            {type} - {city}
            <br />
            {Number(bedrooms)}
            <br />
            {Number(bathrooms)}
            <br />
            {Number(price)} bells
            <br />
            {email}
            <br />
        </div>
    );
};

PropertyCard.propTypes = {
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    bedrooms: PropTypes.number,
    bathrooms: PropTypes.number,
    price: PropTypes.number,
    email: PropTypes.string.isRequired,
};

PropertyCard.defaultProps = {
    bedrooms: 0,
    bathrooms: 0,
    price: 0,
};

export default PropertyCard;
