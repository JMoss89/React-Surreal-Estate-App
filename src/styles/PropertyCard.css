.property-card{
    border: solid purple;
    width: 20vmin;
    height: 15vmin;
}