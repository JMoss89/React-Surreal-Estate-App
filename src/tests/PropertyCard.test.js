import React from 'react';
import { render, screen } from '@testing-library/react';
import PropertyCard from '../components/PropertyCard';

describe('PropertyCard', () => {
    
    test('should have a class name called property-card', () => {
        const { container } = render(<PropertyCard />);
    
        expect(container.firstChild).toHaveClass('property-card');
    });

    test('renders correctly', () => {
        const validInfo = {
            title: 'test title',
            type: 'test type',
            city: 'test city',
            bedrooms: '4',
            bathrooms: '2',
            price: '400000',
            email: 'test@test.com',
        };
        const { asFragment } = render(
            <PropertyCard 
                title={validInfo.title}
                type={validInfo.type}
                city={validInfo.city}
                bedrooms={validInfo.bedrooms}
                bathrooms={validInfo.bathrooms}
                price={validInfo.price}
                email={validInfo.email}
            />
        );
        expect (asFragment()).toMatchSnapshot();
    });
})