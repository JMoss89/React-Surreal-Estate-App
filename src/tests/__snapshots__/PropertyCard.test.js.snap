// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    test title
    <br />
    test type - test city
    <br />
    4
    <br />
    2
    <br />
    400000 bells
    <br />
    test@test.com
    <br />
  </div>
</DocumentFragment>
`;
